examples/feature_demo/compare_images.py

```diff
@@ -0,0 +1,281 @@
+"""
+A Tool to Compare Two Images of Identical Shapes
+================================================
+
+While mostly an internal tool, it does show how one can combine multiple scenes,
+all sharing the same data to display it different ways.
+
+It shows a grid with 4 images:
+- Top Left: Reference image
+- Top Right: Image to compare
+- Bottom Left: Difference between the images (RGB)
+- Bottom Right: Difference between the images (Alpha)
+
+One can pan and zoom, as well as adjust the contrast of the images.
+
+This example can be used as an example of how a custom shader can be created
+in order to discard pixels based on their color.
+"""
+
+# sphinx_gallery_pygfx_docs = 'screenshot'
+# sphinx_gallery_pygfx_test = 'run'
+
+import argparse
+import sys
+from pathlib import Path
+
+import imageio.v3 as imageio
+import numpy as np
+from imgui_bundle import imgui
+from wgpu.gui.auto import WgpuCanvas, run
+from wgpu.utils.imgui import ImguiRenderer
+
+import pygfx as gfx
+from pygfx.renderers.wgpu import register_wgpu_render_function
+from pygfx.renderers.wgpu.shaders.imageshader import ImageShader
+
+parser = argparse.ArgumentParser(description="Image Comparison")
+# Require two optional arguements
+parser.add_argument("images", nargs="*", help="The two images to compare")
+args = parser.parse_args()
+
+if len(args.images) == 2:
+    reference = imageio.imread(args.images[0])
+    image = imageio.imread(args.images[1])
+elif len(args.images) == 0:
+    gfx_examples_screenshot_dir = (
+        Path(__file__).parent.parent / "screenshots"
+    ).resolve()
+    reference = imageio.imread(
+        gfx_examples_screenshot_dir / "validate_blend_weightedplus.png"
+    )
+    image = imageio.imread(gfx_examples_screenshot_dir / "validate_blend_dither.png")
```

Comment:
I picked these as the two most comparable examples that already exist the validation screenshots.